pytest-selenium/pages/content.py

```diff
@@ -161,15 +168,59 @@ class ToolBar(Region):
             By.CSS_SELECTOR,
             "[aria-label='Click to open the Table of Contents']",
         )
+        _search_textbox_desktop_locator = (By.CSS_SELECTOR, '[data-testid="desktop-search-input"]')
+        _search_button_desktop_locator = (By.CSS_SELECTOR, 'button:nth-of-type(2)[value="Search"]')
+        _search_button_mobile_locator = (By.CSS_SELECTOR, '[data-testid="mobile-toggle"]')
 
         @property
         def toc_toggle_button(self):
             return self.find_element(*self._toc_toggle_button_locator)
 
+        @property
+        def search_textbox(self):
+            return self.find_element(*self._search_textbox_desktop_locator)
+
+        @property
+        def search_button(self):
+            return self.find_element(*self._search_button_desktop_locator)
+
+        @property
+        def search_button_mobile(self):
+            return self.find_element(*self._search_button_mobile_locator)
+
         def click_toc_toggle_button(self):
             self.offscreen_click(self.toc_toggle_button)
             return self.page.sidebar.wait_for_region_to_display()
 
+        def click_search(self):
+            self.offscreen_click(self.search_button)
+            return self.page.sidebar.search_sidebar.wait_for_region_to_display()
+
+        def click_search_icon(self):
+            self.offscreen_click(self.search_button_mobile)
+
+        def search_for(self, element):
+            self.search_textbox.send_keys(element)
+            self.click_search()
+            return self.page.sidebar.search_sidebar.wait_for_region_to_display()
+
+    class MobileSearch(Region):
+        _search_textbox_mobile_locator = (By.CSS_SELECTOR, '[data-testid="mobile-search-input"]')
+
+        @property
+        def search_textbox(self):
+            return self.find_element(*self._search_textbox_mobile_locator)
+
+        def trigger_search(self):
+            self.offscreen_click(self.search_textbox)
```

Comment:
clicking on the textbox doesn't trigger anything, what does this function do?

Reply:
I was calling this `trigger_search` in search_for(). Like:
1. enter text in searchbox
2. call trigger_search => hit enter on the search textbox which would trigger the search

Now I have removed the trigger_search and updated `search_for`